Bug report: Comma dangle always throw error

**Tell us about your environment**

* **ESLint Version:** v3.12.2
* **Node Version:** v6.9.1
* **npm Version:** 4.0.5

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```js
{
  "globals": {},
  "env": {},
  "rules": {
    "comma-dangle": [
      "error",
      {
        "arrays": "always-multiline",
        "objects": "always-multiline",
        "imports": "always-multiline",
        "exports": "always-multiline",
        "functions": "always-multiline"
      }
    ]
  },
  "parserOptions": {}
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
console.log(
	'a'
);
```

**What did you expect to happen?**

```sh
$> eslint a.js
# No error
```

**What actually happened? Please include the actual, raw output from ESLint.**

```sh
mondwan@mondwan-All-Series:~/Documents/codeTest/babelEslint$ eslint --print-config .
{
  "globals": {},
  "env": {},
  "rules": {
    "comma-dangle": [
      "error",
      {
        "arrays": "always-multiline",
        "objects": "always-multiline",
        "imports": "always-multiline",
        "exports": "always-multiline",
        "functions": "always-multiline"
      }
    ]
  },
  "parserOptions": {}
}
mondwan@mondwan-All-Series:~/Documents/codeTest/babelEslint$ cat a.js 
console.log(
	'a'
);
mondwan@mondwan-All-Series:~/Documents/codeTest/babelEslint$ eslint a.js 

/home/mondwan/Documents/codeTest/babelEslint/a.js
  2:5  error  Missing trailing comma  comma-dangle

✖ 1 problem (1 error, 0 warnings)


```

--- a/docs/rules/comma-dangle.md
+++ b/docs/rules/comma-dangle.md
@@ -68,8 +68,9 @@ The default for each option is `"never"` unless otherwise specified.
 * `objects` is for object literals and object patterns of destructuring. (e.g. `let {a,} = {a: 1};`)
 * `imports` is for import declarations of ES Modules. (e.g. `import {a,} from "foo";`)
 * `exports` is for export declarations of ES Modules. (e.g. `export {a,};`)
-* `functions` is for function declarations and function calls. (e.g. `(function(a,){ })(b,);`)<br>
-  `functions` is set to `"ignore"` by default for consistency with the string option.
+* `functions` is for function declarations and function calls. (e.g. `(function(a,){ })(b,);`)
+    * `functions` is set to `"ignore"` by default for consistency with the string option.
+    * `functions` should only be enabled when linting ECMAScript 2017 or higher.
 
 ### never
 

